libs/designsystem/button/src/button.component.ts

```diff
@@ -92,10 +93,28 @@ export class ButtonComponent implements AfterContentInit {
   @ContentChild(IconComponent, { read: ElementRef })
   iconElementRef?: ElementRef<HTMLElement>;
 
+  private removeListeners: (() => void)[] = [];
```

Comment:
Name might be slightly confusing, consider renaming and leaving the "removal" stuff as an implementation detail in ngOnDestroy. 

```suggestion
  private eventListeners: (() => void)[] = [];
```